paddle/fluid/framework/ir/graph.cc

```diff
@@ -24,6 +24,68 @@ namespace paddle {
 namespace framework {
 namespace ir {
 
+std::vector<std::string> FindDistTrainSendVars(
+    const std::vector<ir::Node *> &nodes) {
+  std::vector<std::string> send_vars;
+  // since parameters are all in block 0,
+  // it's enough to only scan send ops in block 0
+  for (auto &node : nodes) {
+    auto op_vars = node->Op()->InputArgumentNames();
+    send_vars.reserve(send_vars.size() +
```

Comment:
> However, if we know how many elements are going to be inserted, we can do a reserve before the insertion, so that the vector doesn’t have a reallocate multiple times during insertions.

as it says. call `reserve` in the for loop is not efficient at all.